Reorg docs menu and start adding content (#838)
kubeshop/tracetest: docs/introduction.md

> # Introduction to Trace Based Testing

Reviewer: ```suggestion
# Introduction to Trace-Based Testing

```

> Trace Based Testing is a means of conducting deep integration or system tests by utilizing the rich data contained in a distributed system trace.

Reviewer: ```suggestion
Trace-Based Testing is a means of conducting deep integration or system tests by utilizing the rich data contained in a distributed system trace.

```

## What is a Distributed Trace?

A Distributed Trace, more commonly known as a Trace, records the paths taken by requests (made by an application or end-user) take as they propagate through multi-service architectures, like microservice and serverless applications. [Source - OpenTelemetry.io](https://opentelemetry.io/docs/concepts/observability-primer/)

![Trace & Spans Diagram](img/trace-explainer.gif)

### What is a Span?

Traces are comprised of spans. A span represents a single operation in a trace. Spans are nested, typically with a parent child relationship to form a deeply nested tree.

> ### What data do Spans contain?

Reviewer: ```suggestion
### What Data do Spans Contain?

```

A span contains the data about the operation it represents. This data includes:

> - The span name

Reviewer: ```suggestion
- The span name.

```

> - Start and end timestamp

Reviewer: ```suggestion
- Start and end timestamp.

```

> - List of events (if instrumented)

Reviewer: ```suggestion
- List of events (if instrumented).

```

- Attributes

### What are Attributes?

Attributes are a key-value pair, and they contain information about the operation. A developer can manually add additional attributes to a span, enriching the data. There are [Semantic Conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/) that provide recommended names for the attributes for common types of calls such as database, http, messaging, etc.

> ## What is an assertion?

Reviewer: ```suggestion
## What is an Assertion?

```

> In Tracetest an assertion is comprised of two parts:

Reviewer: ```suggestion
In Tracetest, an assertion is comprised of two parts:

```

- Selector
- Checks

![Selectors and Checks](img/assertion-explainer.gif)

> ### What is a selector?

Reviewer: ```suggestion
### What is a Selector?

```

> A selector contains criteria to limit the scope of the spans from a trace that we wish to assert against. A selector can be very narrow, only selecting on span, or very wide, selecting all spans or all spans of a certain type or other characteristic. Underlying this capability is a [selector language](/docs/installing.md).

Reviewer: ```suggestion
A selector contains criteria to limit the scope of the spans from a trace that we wish to assert against. A selector can be very narrow, only selecting on one span, or very wide, selecting all spans or all spans of a certain type or other characteristics. Underlying this capability is a [selector language](/docs/installing.md).

```

> ### What is a check?

Reviewer: ```suggestion
### What is a Check?

```

A check is a logical verification that will be performed on all spans that match the selector. It is comprised of an attribute, a comparison operator, and a value.

> ### What is a span signature?

Reviewer: ```suggestion
### What is a Span Signature?

```
